### PR TITLE
Fix Virtual Console in /learn

### PR DESCRIPTION
**Pull request by KrisXV**

### CODE REVIEW

**Comment by Zarel** [comment]
I thought you decided to use a new letter for LGPE? Did that change?

File: server/chat-plugins/datasearch.js [hunk 1]
```diff
@@ -1764,7 +1764,11 @@ function runLearn(target, cmd) {
 				return source;
 			}).sort();
 			for (let source of sources) {
-				buffer += `<li>Gen ${source.charAt(0)} ${sourceNames[source.charAt(1)]}`;
```
Comment (Zarel) [inline server/chat-plugins/datasearch.js]: There's no need for five lines when you can just write

```js
buffer += `<li>Gen ${source.charAt(0)} ${sourceNames[source] || sourceNames[source.charAt(1)]}`;
```

File: server/chat-plugins/datasearch.js [hunk 2]
```diff
@@ -1748,7 +1748,7 @@ function runLearn(target, cmd) {
 	buffer += `${template.name}` + (problems ? ` <span class="message-learn-cannotlearn">can't</span> learn ` : ` <span class="message-learn-canlearn">can</span> learn `) + Chat.toListString(moveNames);
 	if (!problems) {
 		let sourceNames = {
-			E: "", S: "event", D: "dream world", V: "virtual console transfer from gen 1-2", X: "traded-back ", Y: "traded-back event", G: "transferred from Let's Go!",
+			'7V': "virtual console transfer from gen 1-2", '8V': "virtual console transfer from LGPE", E: "", S: "event", D: "dream world", X: "traded-back ", Y: "traded-back event",
```
Comment (Zarel) [inline server/chat-plugins/datasearch.js]:
```suggestion
			'7V': "virtual console transfer from gen 1-2", '8V': "Pokémon Home transfer from LGPE", E: "", S: "event", D: "dream world", X: "traded-back ", Y: "traded-back event",
```

Comment (Zarel) [reply server/chat-plugins/datasearch.js]: LGPE transfers aren't virtual console, are they?

Comment (KrisXV) [reply server/chat-plugins/datasearch.js]: Nope

**Comment by KrisXV** [comment]
@Zarel TI did this when fixing the learnsets order and didnt update this file

**Pull request merged**